contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -394,4 +399,78 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
             $.slot := IpRoyaltyVaultStorageLocation
         }
     }
+
+    function claimRevenueOnBehalfByTokenBatch(
+        address[] calldata tokenList,
+        address claimer
+    ) external nonReentrant whenNotPaused returns (uint256[] memory) {
+        IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
+
+        if (ROYALTY_MODULE.isIpRoyaltyVault(claimer) && msg.sender != claimer)
+            revert Errors.IpRoyaltyVault__VaultsMustClaimAsSelf();
+
+        if (IP_ASSET_REGISTRY.isWhitelistedGroupRewardPool(claimer) && msg.sender != GROUPING_MODULE)
+            revert Errors.IpRoyaltyVault__GroupPoolMustClaimViaGroupingModule();
+
+        uint256[] memory claimableAmounts = new uint256[](tokenList.length);
+        for (uint256 i = 0; i < tokenList.length; i++) {
+            claimableAmounts[i] = _clearPendingRewards(claimer, tokenList[i]);
+            if (claimableAmounts[i] == 0) revert Errors.IpRoyaltyVault__NoClaimableTokens();
```

Comment:
Maybe better to just skip to the next token in the token list instead of reverting?